test/unit/test_bigquery_adapter.py

```diff
@@ -287,3 +290,81 @@ def test_invalid_relation(self):
         }
         with self.assertRaises(hologram.ValidationError):
             BigQueryRelation.from_dict(kwargs)
+
+
+class TestBigQueryConnectionManager(unittest.TestCase):
+
+    def setUp(self):
+        credentials = Mock(BigQueryCredentials)
+        profile = Mock(query_comment=None, credentials=credentials)
+        self.connections = BigQueryConnectionManager(profile=profile)
+        self.mock_client = Mock(
+          dbt.adapters.bigquery.impl.google.cloud.bigquery.Client)
+        self.mock_connection = MagicMock()
+
+        self.mock_connection.handle = self.mock_client
+
+        self.connections.get_thread_connection = lambda: self.mock_connection
+
+    def test_retry_and_handle(self):
+        self.connections.DEFAULT_MAXIMUM_DELAY = 2.0
+        dbt.adapters.bigquery.connections._is_retryable = lambda x: True
+
+        @contextmanager
+        def dummy_handler(msg):
+            yield
+
+        self.connections.exception_handler = dummy_handler
+
+        class DummyException(Exception):
+            """Count how many times this exception is raised"""
+            count = 0
+
+            def __init__(self):
+                DummyException.count += 1
+
+        def raiseDummyException():
+            raise DummyException()
+
+#        with self.assertLogs(logger.name) as logs:
+        with self.assertRaises(DummyException):
+            self.connections._retry_and_handle(
+                 "some sql", {'credentials': {'retries': 8}},
```

Comment:
This should be a mock credentials object now, instead of a dict

Reply:
Good catch.